File: android/src/main/java/com/nabbra/rnbluesmanager/EventType.kt
package com.nabbra.rnbluesmanager

import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.WritableMap

enum class EventType(val eventName: String) {
  BLUETOOTH_STATE_CHANGING("bluetoothStateChanging"),
  BLUETOOTH_STATE_CHANGED("bluetoothStateChanged"),
  BLUETOOTH_ENABLED("bluetoothEnabled"),
  BLUETOOTH_DISABLED("bluetoothDisabled"),
  CONNECTION_STATE_CHANGED("connectionStateChanged"),
  DEVICE_CONNECTED("deviceConnected"),
  DEVICE_DISCONNECTED("deviceDisconnected"),
  ERROR("error"),
  DEVICE_DISCOVERED("deviceDiscovered"),
  SCAN_STARTED("scanStarted"),
  SCAN_STOPPED("scanStopped");
}

File: ios/EventType.swift
import Foundation

@objc
enum EventType: Int {
    case bluetoothStateChanging
    case bluetoothStateChanged
    case bluetoothEnabled
    case bluetoothDisabled
    case connectionStateChanged
    case deviceConnected
    case deviceDisconnected
    case error
    case deviceDiscovered
    case scanStarted
    case scanStopped

    var eventName: String {
        switch self {
        case .bluetoothStateChanging: return "bluetoothStateChanging"
        case .bluetoothStateChanged: return "bluetoothStateChanged"
        case .bluetoothEnabled: return "bluetoothEnabled"
        case .bluetoothDisabled: return "bluetoothDisabled"
        case .connectionStateChanged: return "connectionStateChanged"
        case .deviceConnected: return "deviceConnected"
        case .deviceDisconnected: return "deviceDisconnected"
        case .error: return "error"
        case .deviceDiscovered: return "deviceDiscovered"
        case .scanStarted: return "scanStarted"
        case .scanStopped: return "scanStopped"
        }
    }
}
